src/opendap/auth/IdFilter.java

```diff
@@ -454,7 +454,12 @@ else if (enableGuestProfile && requestURI.equals(guestEndpoint)) {
                         try {
                             retVal = IdPManager.getDefaultProvider().doTokenAuthentication(request, userProfile);
                         } finally {
-                            logEDLProfiling("Validate token - Is valid? " + retVal, profilingStartTime);
+                            // We only care about logging token validation if there was ever a token to BE validated.
+                            // If there was no userProfile, there was never even the hint of a token to be validated,
+                            // and `doTokenAuthentication` will have returned immediately.
+                            if (userProfile != null) {
```

Comment:
Can `userProfile` ever be null here? 
* I don't think so... It's declared and assigned  on line 451. And while `doTokenAuthentication()` does check for a null `user_profile` reference I don't think that can ever happen in this code block.

So maybe a different test? Is the state of `retVal` indeterminate after an exception, or is it only going to be true is `doTokenAuthentication()` completes?
Or maybe check something in the `user_profile` that gets set when there is a successful `doTokenAuthentication()` call?

Reply:
ah interesting, good point. well, glad to know that now instead of see it once I get to SIT! I will rethink this.

Reply:
Okay, I think I fixed this.